src/transaction.js

```diff
@@ -229,7 +229,7 @@ function makeTxClient(trx, client, connection) {
     })
   }
   trxClient.acquireConnection = function() {
-    return Promise.settle([trx._previousSibling]).then(function () {
+    return Promise.all([trx._previousSibling].map((p) => { p.reflect() })).then(function () {
```

Comment:
Here probably you should return promise in map?

`.map((p) => { return p.reflect() })` or `.map((p) => p.reflect())` otherwise you will get `Promise.all([undefined])` here.

So maybe 

`return Promise.all([trx._previousSibling].map(p => p.reflect())).then(function () { return connection; });`

Also since code already requires that `trx._previousSibling` is a Promise or otherwise p.reflect() would throw an error, so this could be further simplified:

`return trx._previousSibling.reflect().then(function () { return connection; });`